yt/yt/server/controller_agent/controllers/sort_controller.cpp

```diff
@@ -2576,7 +2577,8 @@ class TSortControllerBase
     {
         if (SimpleSort) {
             UpdateTask(SimpleSortTask);
-        } else {
+        }
+        if (IntermediateSortTask) {
```

Comment:
It is rather suspicioius, that `UpdateTask(FinalSortTask)`  goes inside this if..

Reply:
Fixed. Btw, there is a common pattern
```
if (SomeTask) {
    UpdateTask(SomeTask);
}
```
in controllers. What do you think about refactoring (in a separate PR) allowing UpdateTask to handle nullptr?

Reply:
Soungs good!)